http-protocol-binding.md

```diff
@@ -213,16 +213,75 @@ and per [RFC7230, section 3][rfc7230-section-3], HTTP headers MUST only use
 printable characters from the US-ASCII character set, and are terminated by a
 CRLF sequence with OPTIONAL whitespace around the header value.
 
-String values MUST be percent-encoded as described in [RFC3986, section
-2.4][rfc3986-section-2-4] before applying the header encoding rules described in
-[RFC7230, section 3.2.6][rfc7230-section-3-2-6].
-
-When decoding an HTTP message into a CloudEvent, these rules MUST be applied in
-reverse -- [RFC7230, section 3.2.6][rfc7230-section-3-2-6] decoding to an ASCII
-string, and then a **single round** of percent-decoding as described in
-[RFC3986, section 2.4][rfc3986-section-2-4] to produce a valid UTF-8 String.
-(Note that applying percent-decoding an incorrect number of times can result in
-message corruption or security issues.)
+When encoding a CloudEvent as an HTTP message, string values
+represented as HTTP header values MUST be percent-encoded as
+described below. This is compatible with [RFC3986, section
+2.1][rfc3986-section-2-1] but is more specific about what needs
+encoding. The resulting string SHOULD NOT be further encoded.
+(Rationale: quoted string escaping is unnecessary when every space
+and double-quote character is already percent-encoded.)
+
+When decoding an HTTP message into a CloudEvent, any HTTP header
+value MUST have quoted string escaping as described in [RFC7230,
+section 3.2.6][rfc7230-section-3-2-6] applied in reverse, and then a
```

Comment:
Might be ok, but the phrase "MUST have quoted string escaping ... applied in reverse" might be hard for some people to parse - it kind of sounds like we're saying it needs to be escaped but then it might not be clear what it means to say "in reverse".  So it's then odd since we can't have a MUST that applies to the sender if we're talking about the receiver/decoder. Yes I know we're just really trying to say they MUST de-escape - so perhaps we can rephrase this to be more clear?

Reply:
I've had a go - see what you think. (I'm very happy to go round on this a few times. I don't want to leave any room for SDK implementers to be unclear about what they need to do.)